uportal-war/src/main/java/org/jasig/portal/groups/pags/dao/jpa/JpaPersonAttributesGroupDefinitionDao.java

```diff
@@ -86,64 +117,37 @@ public void deletePersonAttributesGroupDefinition(IPersonAttributesGroupDefiniti
         entityManager.remove(persistentDefinition);
     }
 
-    @PortalTransactional
     @Override
     public Set<IPersonAttributesGroupDefinition> getPersonAttributesGroupDefinitionByName(String name) {
-        CriteriaBuilder criteriaBuilder = this.getEntityManager().getCriteriaBuilder();
-        CriteriaQuery<PersonAttributesGroupDefinitionImpl> criteriaQuery = 
-                criteriaBuilder.createQuery(PersonAttributesGroupDefinitionImpl.class);
-        Root<PersonAttributesGroupDefinitionImpl> root = criteriaQuery.from(PersonAttributesGroupDefinitionImpl.class);
-        ParameterExpression<String> nameParameter = criteriaBuilder.parameter(String.class);
-        criteriaQuery.select(root).where(criteriaBuilder.equal(root.get("name"), nameParameter));
-        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.getEntityManager().createQuery(criteriaQuery);
+        TypedQuery<PersonAttributesGroupDefinitionImpl> query = this.createCachedQuery(groupDefinitionByNameQuery);
         query.setParameter(nameParameter, name);
-        Set<IPersonAttributesGroupDefinition> groups = new HashSet<IPersonAttributesGroupDefinition>();
-        for (IPersonAttributesGroupDefinition group: query.getResultList()) {
-            groups.add(group);
-        }
+        Set<IPersonAttributesGroupDefinition> groups = new HashSet<IPersonAttributesGroupDefinition>(query.getResultList());
         if(groups.size() > 1) {
             logger.error("More than one PAGS Group found for name: {}", name);
         }
         return groups;
     }
 
-    @PortalTransactional
```

Comment:
Do we need @PortalTransactionalReadOnly here?  (I'm pretty sure we don't need @PortalTransactional).